products/views.py

```diff
@@ -1,3 +1,41 @@
-from django.shortcuts import render
+import json
 
-# Create your views here.
+from django.http  import JsonResponse
+from django.views import View
+from django.db    import transaction, IntegrityError
+
+from products.models import *
+from products.utils  import login_decorator
+
+class ProductDetailView(View):
+    def get(self, request):
+        try:
+            data     = json.loads(request.body)
+            product  = Product.objects.get(id = data['product_id'])
+            sizes_db = Size.objects.all()
+            
+            sizes =[{ 
+                'id'   : size_db.id,
+                'size' : size_db.size
+            } for size_db in sizes_db ]
+
+            product_color_images = ProductColorImage.objects.filter(product_id = data['product_id'])
```

Comment:
이미 위에서 상품 객체를 선언했기 때문에, ProductColorImage class를 통해서 값을 가져오지 않고, 
`product.productcolor..`와 같이 product 객체의 속성으로 접근해서 값을 가져오도록 수정해주세요.

```suggestion
            product_color_images = product...
```